lib/src/firestore_cache.dart

```diff
@@ -77,19 +79,31 @@ class FirestoreCache {
   /// You can also pass in [localCacheKey] as the key for storing the last local
   /// cache date, and [isUpdateCacheDate] to set if it should update the last
   /// local cache date to current date and time.
-  static Future<QuerySnapshot<Map<String, dynamic>>> getDocuments({
-    required Query<Map<String, dynamic>> query,
-    required DocumentReference<Map<String, dynamic>> cacheDocRef,
+  ///
+  /// If you are using firestore collection withConverter then you have to also
+  /// pass toFirestore method to converter
+  static Future<QuerySnapshot<T>> getDocuments<T>({
+    required Query<T> query,
+    required DocumentReference<T> cacheDocRef,
```

Comment:
Same question as above. I wonder if changing `Query<T>` is good enough for the issue you described?